lib/perl/Genome/InstrumentData/AlignmentResult.pm

```diff
@@ -1919,5 +1919,48 @@ sub add_to_temporary_input_files_queue {
     return 1;
 }
 
+sub lock_bam_file_access {
+    my $self = shift;
+    unless ($self->get_merged_alignment_results) {
+        my @bams = $self->alignment_bam_file_paths;
+        unless (@bams) {
+            die $self->error_message("Alignment result with class (%s) and id (%s) has neither ".
+                "merged results nor valid bam paths. This likely means that this alignment result ".
+                "needs to be removed and realigned because data has been lost. ".
+                "Please create an apipe-support ticket for this.", $self->class, $self->id);
+            #There is no way to recreate per lane bam if merged bam
+            #does not exist and per lane bam is removed. Software
+            #result of this per lane alignment needs to be removed and
+            #this per lane instrument data needs to be realigned
+            #with that aligner.
+        }
+
+        my $lock_var = File::Spec->join('genome', __PACKAGE__, 'lock-per-lane-alignment-'.$self->id);
```

Comment:
During the transition to this code, there may be two lock names, since __PACKAGE__ used to be the merged alignment result and is now alignment result.  We may need to take steps to mitigate this?